File: Java/Medium/Accepted/adding-spaces-to-a-string_6.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        
        StringBuilder sb = new StringBuilder();
        int n = spaces.length;
        int j = 0;
        for (int i = 0; i < s.length(); i++) {
            if (j < n && i == spaces[j]) {
                sb.append(' ');
                j++;
            }
            sb.append(s.charAt(i));
        }
        return sb.toString();
    }
}

File: Rust/Medium/Accepted/adding-spaces-to-a-string_6.rs
impl Solution {
    pub fn add_spaces(s: String, spaces: Vec<i32>) -> String {
        
        let mut res = String::new();
        let mut i = 0;
        for &space in spaces.iter() {
            res += &s[i..space as usize];
            res += " ";
            i = space as usize;
        }
        res += &s[i..];
        res
    }
}